src/codemodder/registry.py

```diff
@@ -0,0 +1,175 @@
+from dataclasses import dataclass, asdict
+from importlib.resources import files
+from importlib.metadata import entry_points
+from typing import Optional
+
+from codemodder.logging import logger
+
+
+@dataclass
+class CodemodCollection:
+    """A collection of codemods that all share the same origin and documentation."""
+
+    origin: str
+    docs_module: str
+    semgrep_config_module: str
+    codemods: list
+
+
+class _CodemodContainer:
```

Comment:
This is a slightly awkward construct but I might suggest that in the longer time this actually outlines the API we want for the `BaseCodemod` class. Eventually maybe this becomes an ABC that all codemods inherit.

Reply:
More thoughts, since you have the idea of this class `_CodemodContainer` being the "real" BaseCodemod, what do you think is the distinction between the data a BaseCodemod class should store versus this new `_CodemodContainer`? I still feel they should be one class but I'm listening!

Reply:
* I did think about making `CodemodCollection` a `list` but I'm not sure whether we really need that API. It might be worth a quick experiment though
* I do think maybe `_CodemodWrapper` is a better name and it would be a good dataclass
* The main difference between `_CodemodContainer/Wrapper` and `BaseCodemod` is that the container knows about the registry and associated metadata and the base codemod does not. But maybe there's a good way to consolidate those  two things somehow. I think that's a future effort though.